src/algorithms/crypto/randomness.md

```diff
@@ -10,54 +10,89 @@ We describe this formatting below.
+- liveness for leader election -- in the case of null rounds, the new epoch number will output new randomness for LE
+- distinct values for randomness sought before genesis -- where the genesis ticket will be returned
+
+For instance, if in epoch `curr`, a miner wants randomness from `lookback` epochs back where `curr - lookback <= genesis`, 
+the ticket randomness drawn would be `H(genesisTicket.digest || curr-lookback)` where the `genesisTicket` is the randomness included
+in the genesis block (to be determined ahead of time to enable genesis participants to SEAL data ahead of time).
+
+While the inclusion of the round number is not necessary for other uses of entropy, we include it as part of ticket drawing for uniformity/simplicity in the overall protocol.
+
+See the `RandomnessAtEpoch` method below:
+{{< readfile file="../struct/chain/chain.go" code="true" lang="go" >}}
 
 ## Forming Randomness Seeds
 
 Different uses of randomness require randomness seeds predicated on a variety of inputs. For instance, we have:
 
-- `TicketDrawing` -- uses ticket and epoch number
 - `TicketProduction` -- uses ticket and miner actor addr
-- `PoStChallengeSeed` -- uses ticket and miner actor addr
-- `SurprisePoStSelectMiners` -- uses ticket and epoch number
+- `ElectionPoStChallengeSeed` -- uses ticket and miner actor addr
+- `WindowedPoStChallengeSeed` -- uses ticket and epoch number
 ...
 
-In all cases, a ticket is used as the base of randomness (see {{<sref tickets>}}). In order to make randomness seed creation uniform, the protocol derives all such seeds in the same way, as follows (also see {{<sref tickets>}}):
+In all cases, a ticket is used as the base of randomness (see {{<sref tickets>}}). In order to make randomness seed creation uniform, the protocol derives all such seeds in the same way, using blake2b as a hash function to generate a 256-bit output as follows (also see {{<sref tickets>}}):
+
+In round `n`, for a given randomness lookback `l`, and serialized entropy `s`:
+
 ```text
- For a given ticket's randomness ticket_randomness:
+ticket = DrawRandomness(n-l)
+randSeed = ticket.digest
 
 buffer = Bytes{}
 buffer.append(IntToBigEndianBytes(AppropriateDST))
-buffer.append(-1) // a flag to be used in cases where FIL might need longer randomness outputs. Currently unused
-buffer.append(ticket_randomness)
-buffer.append(other needed serialized inputs)
+buffer.append(randSeed)
+buffer.append(s)
 
-randomness = SHA256(buffer)
+randomness = H(buffer)
 ```
 
 {{< readfile file="/docs/actors/actors/crypto/randomness.go" code="true" lang="go" >}}
 {{< readfile file="/docs/systems/filecoin_blockchain/struct/chain/chain.go" code="true" lang="go" >}}
+
+## Entropy to be used with randomness
+
+We currently distinguish the following entropy needs per use:
+
+- TicketProduction: requires MinerIDAddress
+- ElectionPoStChallengeSeed: requires current epoch and MinerIDAddress
```

Comment:
This line misled me into thinking the epoch needed to be explicitly mixed in again, even though it's already mixed in  as described at line 49

Reply:
Including ticket production and epost challenge seed here _at all_ misled me into thinking they could be calculated the same way. I suggest removing them entirely and then describing how they can be computed, given ticket-digest-plus-epoch described above (or, more importantly, if they diverge even lower down).

Reply:
No but some of this is stale. Thank you!

Let me clean up.